axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+| `MessageNameSpanAttributesProvider`         | `axon_message_name`         | The name of the message for Commands and Queries                                        |
+| `MessageTypeSpanAttributesProvider`         | `axon_message_type`         | The class of the message, such as `DomainEventMessage` or `GenericQueryMessage`         |
+| `PayloadTypeSpanAttributesProvider`         | `axon_payload_type`         | The class of the payload in the message                                                 |
+| `MetadataSpanAttributesProvider`            | `axon_metadata_{key}`       | All metadata of the message is also added to the span with its corresponding key        |
+
+In addition to the ones provided by the framework, you can also create a custom `SpanAttributesProvider`
+and add it to the `SpanFactory`, in case you want to add custom information you would like to include as a label.
+
+```java
+public class CustomSpanAttributesProvider implements SpanAttributesProvider {
+
+    @Nonnull
+    @Override
+    public Map<String, String> provideForMessage(@Nonnull Message<?> message) {
+        // Provide your labels based on the message here
+        return Collections.emptyMap();
+    }
+}
+```
+
+You can register this custom `SpanAttributesProvider` in one of the following ways.
+
+{% tabs %}
+{% tab title="Spring Boot Bean" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Bean
+    public SpanAttributesProvider customSpanAttributesProvider() {
+        // Autoconfiguration picks beans of type SpanAttributesProvider up automatically.
+        return new CustomSpanAttributesProvider();
+    }
+}
+```
+
+{% endtab %}
+{% tab title="Spring Boot injection" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Autowired
+    public void configureSpanFactory(SpanFactory spanFactory) {
+        spanFactory.registerSpanAttributeProvider(new CustomSpanAttributesProvider());
+    }
+}
+```
+
+{% endtab %}  
+{% tab title="Manual" %}
+
+```java
+public class AxonConfigurer {
+
+    public void configure(Configuration configuration) {
+        configuration.spanFactory().registerSpanAttributeProvider(new CustomSpanAttributesProvider());
+    }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## OpenTelemetry <a id="opentelemetry"></a>
+
+Axon work provides [OpenTelemetry support](https://opentelemetry.io/docs/concepts/what-is-opentelemetry/) out of the
+box. The OpenTelemetry standard improves upon the OpenTracing standard by providing more
+auto-instrumentation without the need for the user to configure many things.
+
+OpenTelemetry works by adding a Java agent to the execution of the application. Based on the configuration, the agent
+will collect logs, metrics and tracing automatically before sending it to a collector that can provide insights.
+ElasticSearch APM, Jaeger and many other tools are available for this and the configuration of these is beyond the scope
```

Comment:
I assume that "this" references to collecting insights?

Reply:
Adjusted wording here